core/wire/commands/commands.go

```diff
@@ -216,14 +221,14 @@ func getConsensusFromBytes(b []byte) (Command, error) {
 // GetVote is a de-serialized get_vote command.
 type GetVote struct {
 	Epoch     uint64
-	PublicKey *eddsa.PublicKey
+	PublicKey sign.PublicKey
 }
 
 // ToBytes serializes the GetVote and returns the resulting slice.
 func (v *GetVote) ToBytes() []byte {
 	out := make([]byte, cmdOverhead+8, cmdOverhead+voteOverhead)
```

Comment:
not related to this PR, but `GetVote` emits a bunch of stuff that `getVoteFromBytes` doesn't parse at the beginning. I'm assuming the parsing is split across two functions somewhere. Is this intentional? Wonder we don't just use CBOR for this.
as far as I can see `cmdOverhead = 1+1+4` and `voteOverhead = 8 + PublicKeySize`, and the byte layout seems to be:
```
offset
0 : command
1 : unspecified
2: 4 bytes of voteOverhead
---- getVoteFromBytes deals only with this:
6: 8 bytes of epoch
14: whatever bytes of public key
```
do we know what `out[1]` is supposed to hold? is `command` a little-endian uint16, and the other fields are big-endian?

Reply:
this hasn't changed that much in this PR.

Reply:
no, it's unrelated to that this PR really.